neevaco/neeva-ios
# Add Creating jpeg Data Using CGImageDestination

As listed on the issue, we cannot save thumbnails of pdfs because of unsupported pixel format in the UIImage. 
```
writeImageAtIndex:756: *** CMPhotoCompressionSessionAddImage: err = kCMPhotoError_UnsupportedPixelFormat [-16995]    (codec: 'jpeg')
```

We also identified that the cgImage object is still valid. As a workaround, we using CGImageDestination as a fallback method to create the jpeg data.

## Checklists

### How was this tested?

### Manual test cases
Open a link to pdf, open another tab. look at them in the tab switcher, open a tab again so the preserveTab function gets triggered. close the app and reopen. Verify in tab switcher.

### Screenshots and screen recordings

https://user-images.githubusercontent.com/44557675/157605902-dd91234a-3e5c-4ded-93ad-5c4a74dbb1b7.MP4



## Issues addressed
resolves #3001
